crates/moonbuild/src/entry.rs

```diff
@@ -600,15 +602,41 @@ pub fn run_test(
     }
 
     let res = if moonbuild_opt.no_parallelize {
+        let runtime = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()?;
+
         runtime.block_on(async {
             let mut results = vec![];
             for handler in handlers {
+                // Tasks are run sequentially by using the `await` expression directly.
                 results.push(handler.await);
             }
             results
         })
     } else {
-        runtime.block_on(futures::future::join_all(handlers))
+        let thread_count = match std::thread::available_parallelism() {
+            Ok(n) => n.get(),
+            Err(_) => num_cpus::get(),
```

Comment:
could we just set `thread_count` to 1 in here, so we can remove `num_cpus` dep

cc @lijunchen

Reply:
I notice the default worker thread count in `tokio` is the number of CPUs and it also uses the method `std::thread::available_parallelism`. If `std::thread::available_parallelism` fails, the thread count will default to `1`. The related invocation link is shown below.
```
tokio::runtime::builder::Builder::build ->
tokio::runtime::builder::Builder::build_threaded_runtime ->
tokio::loom::sys::num_cpus
```

> could we just set thread_count to 1 in here, so we can remove num_cpus dep

So I agree with your idea and propose not to set the worker thread count munually (just use the default value in `tokio`).